### DISCUSSION TITLE
[Question][Hw3] coverage of phi node + necessity of functions in Utils

### DISCUSSION THREAD

**Discussion by AnHaechan** [prosyslab-classroom/cs348-information-security#551] [Q&A]
I found out the provided function `Utils.debug_location` returns None when PHI instruction is fed as its argument.

But according to the specification of the homework, we need to instrument `__coverage__` for **every instruction except instruction s.t. `Utils.is_debug instruction = true`**, and I also checked the PHI node is really a 'instruction' In LLVM IR, from the LLVM document.

 **(1)** Then, should we somehow get debug information of phi node without depending on the provided function in `Utils`?

I have one more question related to module `Utils`.
I really thought hard to come up with the reason why do I need some functions in `Utils` such as `find_main`, `is_print`, `is_unary_op`, `string_of_exp` but it was difficult to find out their usage.
For `find_main`, `is_print`, `is_input`, I don't understand why should we consider specific functions.
For `is_assignment`, `is_unary_op`, `is_binary_op`, I don't think method of inserting `__coverage__` will differ because of the kind of instrunction..

**(2)**  Are these things really necessary for good sanitizer?
If so, can you spare me some hints..?

Thanks.

**Comment by KihongHeo** [comment]
You can ignore all instructions such that Utils.debug_location returns None.
